include/seastar/util/log.hh

```diff
@@ -213,24 +221,49 @@ public:
     ///              or a logger::format_info passed down the call chain.
     /// \param args - args to print string
     ///
-    template <typename... Args>
-    void log(log_level level, format_info fmt, Args&&... args) noexcept {
+    template <typename Fmt, typename... Args>
+#if FMT_VERSION >= 80000
+    [[deprecated("Use log(log_level, fmt::format_string<Args...>, Args&&...) instead")]]
+    std::enable_if_t<std::is_same_v<Fmt, format_info>>
```

Comment:
I don't understand what I'm seeing here. @bhalevy I think you added the format_info thing in commit 2cd0a2928782b89c71d16bc2d2628d987a17e874 - can you please check that you're ok with it being deprecated (if that's in fact what is happening here...)

Reply:
the change here is based on by my understanding of this function: i think the point of the try-catch block is to let the user be aware of the code location where we have bad format string or just the place where customized formatter throws. so ideally, the compile-time format string check should help with this use case. and i believe the customized formatter *should* not throw.

but to be 100% compatible with existing code base which *might* created a `format_info` instance manually and pass it to `log()`, and apparently `format_info` is not a `consteval` `string_view` from which a `fmt::format_string<Args...>` instance can be instantiated. the existing function is preserved. but to avoid ambiguity between the new `log(log_level, fmt::format_string<Args...>, Args&&...)` and the `format_info` version, i have to teach the compiler to pick the new one as long as the 2nd parameter can be converted to `fmt::format_string<Args...>` .

Reply:
> i think the point of the try-catch block is to let the user be aware of the code location where we have bad format string or just the place where customized formatter throws

_Or_ where memory allocation failed and left logging message not shown

Reply:
indeed, that could happen if the formatted argument(s) expand into a large chunk of string. i need rethink and reevaluate the solutions preserving the source location. the compile-time format checking is quite handy, and i really want to enable it in Seastar logging subsystem if possible.

Reply:
okay, i gave up on enabling the compile-time format checking in Release build. but we can still have it in the Debug build.

Reply:
The formatter can fail as well.

Reply:
i assume that you are referring the formatter of `saved_backtrace`.   please note that both formatter can fail: the one which formats `saved_backtrace` and the one formats `format_info`.  but i agree there are better chances to fail in comparison when formatting `format_info`. in hope to reduce the probability of the failures, how about formatting just a single frame of the backtrace?